src/Life/Main/Add.hs

```diff
@@ -45,28 +51,51 @@ lifeAdd lPath = whatIsLife >>= \case
                 filePath <- resolveFile homeDirPath path
                 whenM (doesFileExist filePath) $ do
                     relativeFile <- makeRelative homeDirPath filePath
-                    resolveConfiguration singleFileConfig filename relativeFile
+                    resolveConfiguration files checkEqualFiles copyFileWithDir relativeFile
 
             (Dir path)  -> do
                 dirPath <- resolveDir homeDirPath path
                 whenM (doesDirExist dirPath) $ do
                     relativeDir <- makeRelative homeDirPath dirPath
-                    resolveConfiguration singleDirConfig dirname relativeDir
+                    resolveConfiguration directories checkEqualDirs copyDirRecur relativeDir
 
             -- We didn't find the file
         errorMessage "The file/directory doesn't exist" >> exitFailure
 
-
-resolveConfiguration :: (Path Rel t -> LifeConfiguration)
-                     -> (Path Rel t -> Path Rel t)
+resolveConfiguration :: Lens' LifeConfiguration (Set (Path Rel t))
+                     -> (Path Rel t -> IO Bool)
+                     -> (Path Abs t -> Path Abs t -> IO ())
                      -> Path Rel t
                      -> IO ()
-resolveConfiguration configBuilder pathName path = do
+resolveConfiguration confLens checkContent copyFun path = do
     configuration <- parseGlobalLife
-    let newConfiguration = configuration <> configBuilder path
-    writeGlobalLife newConfiguration
+    let newConfiguration = configuration & confLens %~ Set.insert path
+
+    isSameAsInRepo <- checkContent path
```

Comment:
I don't see how you check if this file was already in the repo, because I guess checkContent may fail in that case..

Reply:
You can just compare configs for that

Reply:
@vrom911 Good point. Need to fix `isSameAsInRepo` function to check whether file exist.